server/methods/getRoomIdByNameOrId.js

```diff
@@ -17,7 +17,7 @@ Meteor.methods({
 			});
 		}
 
-		if (room.t !== 'c' || RocketChat.authz.hasPermission(Meteor.userId(), 'view-c-room') !== true) {
+		if (RocketChat.authz.hasPermission(Meteor.userId(), `view-${ room.t }-room`) !== true) {
```

Comment:
you cannot do this. the previous check works because everyone can access a public channel. but the same doesn't apply for private groups.

Reply:
not everyone has the right to access public channels, only those with the proper `view-c` permission. 
The change will still work for public channels, but in addition return Ids of non-public-channels - if the user has permission to view them. 
Is there a reason why a user should be allowed to get the Ids of channels for which he’s not authorized if he’s authorized to see public channels? 🧐

Reply:
I don't think I have totally understood your question 😅 

but so far we do not show to users a list of private channels/groups, people also cannot "guess" a private group's name, we don't tell them if that group exists or not.. if someone tries to guess a private group name from the URL, he receives a message like `"No private group with name "channel-name" was found!"`

So we need to be consistent here and not give them the private group's `_id`

Reply:
I'd like to flag I also have a pending PR for related behaviour. #11703

[My commit detailed here.](https://github.com/RocketChat/Rocket.Chat/pull/11703/commits/c42274b8dd52fcf39cc27a5ddf8a5b5209602499)

@sampaiodiego Would be great if you can look at that to find a solution in balance with this one.

The changes to the `getRoomIdByNameOrId` method knocked out a bunch of people's bots, because the bot can no longer get the ID for a private room, even when the bot is in the room.

See #11552